x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java

```diff
@@ -42,12 +41,21 @@ public class LocalLogicalPlanOptimizer extends ParameterizedRuleExecutor<Logical
 
     private final LogicalVerifier verifier = LogicalVerifier.LOCAL_INSTANCE;
 
-    private static final List<Batch<LogicalPlan>> RULES;
-
-    static {
-
-        RULES = arrayAsArrayList(localRewrites(), localOperators(), localCleanup());
-    }
+    private static final List<Batch<LogicalPlan>> RULES = arrayAsArrayList(
+        new Batch<>(
+            "Local rewrite",
+            Limiter.ONCE,
+            new IgnoreNullMetrics(),
+            new ReplaceTopNWithLimitAndSort(),
+            new ReplaceFieldWithConstantOrNull(),
+            new InferIsNotNull(),
+            new InferNonNullAggConstraint(),
+            new ReplaceDateTruncBucketWithRoundTo(),
+            new PushExpressionsToFieldLoad()
```

Comment:
Add the PushExpressionsToFieldLoad rule by default - now the `BlockLoaderExpression.tryPushToFieldLoading` method needs to take into account pushdown capabilities